src/services/vehicleDragger.ts

```diff
@@ -32,13 +32,15 @@ export function toggleVehicleDragger(isPressed: boolean, storeVehicle: Store<[Ri
 	}
 
 	const multiplayer = isMultiplayer();
+	const trackLocation = storeTrackLocation.get();
+	const trackElementIndex = trackLocation ? findTrackElementIndex(trackLocation) : null;
 	const originalPosition =
 	{
 		revert: true,
-		x: storeX.get(),
-		y: storeY.get(),
-		z: storeZ.get(),
-		track: storeTrackLocation.get(),
+		x: trackLocation?.x ?? storeX.get(),
+		y: trackLocation?.y ?? storeY.get(),
+		z: trackLocation?.z ?? storeZ.get(),
```

Comment:
I'd like to avoid `?.` and `??` as they create kind of verbose code when transpiled to ES5. I think it'd be possible to have something concise without the use of these, but I can also do that myself if you don't want to be bothered with those details. 🙂

Reply:
```
	const originalPosition =
	{
		revert: true,
		x: trackLocation && trackLocation.x ? trackLocation.x : storeX.get(),
		y: trackLocation && trackLocation.y ? trackLocation.y : storeY.get(),
		z: trackLocation && trackLocation.z ? trackLocation.z : storeZ.get(),
		trackElementIndex: trackElementIndex,
		progress: storeTrackProgress.get()
	};
```
?

Reply:
What about something like this?

```ts
	const originalPosition = { revert: true };

	if (trackLocation)
	{
		originalPosition.x = trackLocation.x;
		originalPosition.y = trackLocation.y;
		originalPosition.z = trackLocation.z;
		originalPosition.trackElementIndex = findTrackElementIndex(trackLocation);
		originalPosition.progress = storeTrackProgress.get();
	}
	else
	{
		originalPosition.x = storeX.get();
		originalPosition.y = storeY.get();
		originalPosition.z = storeZ.get();
	}
```

Reply:
yessir